### PR TITLE
Update the semconv to spec v1.4.0

### PR DESCRIPTION
**Pull request by jkwatson**

### CODE REVIEW

**Comment by anuraaga** [comment]
@jkwatson Any thoughts on releasing SDK 1.3.0 before this or doesn't matter at all?

**Comment by jkwatson** [comment]
> @jkwatson Any thoughts on releasing SDK 1.3.0 before this or doesn't matter at all? Since we're planning to anyways, it allows the versions to be aligned for a bit more.

I'm fine either way. We're always going to be at least one minor version behind the spec (and even if we merge this, there is still the schema stuff to get in).

**Review by anuraaga — APPROVED** [review]

**Review by kubawach — APPROVED** [review]

**Review by arminru — APPROVED** [review]

File: semconv/src/main/java/io/opentelemetry/semconv/trace/attributes/SemanticAttributes.java [hunk 1]
```diff
@@ -638,6 +638,34 @@
...
+   * specify this, the value can be omitted.
+   */
+  public static final AttributeKey<String> RPC_JSONRPC_VERSION = stringKey("rpc.jsonrpc.version");
+
+  /**
+   * `method` property from request. Unlike `rpc.method`, this may not relate to the actual method
+   * being called. Useful for client-side traces since client does not know what will be called on
+   * the server.
+   */
+  public static final AttributeKey<String> RPC_JSONRPC_METHOD = stringKey("rpc.jsonrpc.method");
```
Comment (Oberon00) [inline semconv/src/main/java/io/opentelemetry/semconv/trace/attributes/SemanticAttributes.java:652]: That new attribute is a bit unfortunate, I opened a spec PR: open-telemetry/opentelemetry-specification#1748 to remove it / clarify that rpc.method can be used instead.

**Review by Oberon00 — APPROVED** [review]

File: semconv/src/main/java/io/opentelemetry/semconv/trace/attributes/SemanticAttributes.java (hunk 1 above)
Comment (jkwatson) [reply semconv/src/main/java/io/opentelemetry/semconv/trace/attributes/SemanticAttributes.java:652]: @Oberon00 do you think we should just wait for that to settle before merging this, then?

Comment (Oberon00) [reply semconv/src/main/java/io/opentelemetry/semconv/trace/attributes/SemanticAttributes.java:652]: IMHO it's better to have this one bad attribute than not to update at all.

Comment (jkwatson) [reply semconv/src/main/java/io/opentelemetry/semconv/trace/attributes/SemanticAttributes.java:652]: I just cut the 1.3.0 release, so we'll get what we get in the next version. :)

**Comment by jkwatson** [comment]
@anuraaga @Oberon00 I amended this PR to also populate a new field for the schema URL in each of the files. Please be sure to review again if you have feelings.

**Review by anuraaga — APPROVED** [review]

**Pull request merged**